mail_compose_select_lang/models/report.py

```diff
@@ -0,0 +1,18 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# Copyright 2015 Pedro M. Baeza
+# For license notices, see __openerp__.py file in root directory
+##############################################################################
+from openerp import models, api
+
+
+class Report(models.Model):
+    _inherit = 'report'
+
+    @api.model
+    def translate_doc(self, doc_id, model, lang_field, template, values):
```

Comment:
I am sorry dude, but without test it is not feasible for me, you will face horizontal dependencies issues here.

Look a simple:

```shell
 $: rgrep translate_doc . --include=*.py
```
Returns this method overwritten twice.

```shell
./odoo/addons/report/models/abstract_report.py:62:        # Used in template translation (see translate_doc method from report model)
./odoo/addons/report/models/report.py:93:    def translate_doc(self, cr, uid, doc_id, model, lang_field, template, values, context=None):
./odoo/addons/report/models/report.py:97:        <t t-raw="translate_doc(doc_id, doc_model, 'partner_id.lang', account.report_invoice_document')"/>
./odoo/addons/report/models/report.py:134:        def translate_doc(doc_id, model, lang_field, template):
./odoo/addons/report/models/report.py:135:            return self.translate_doc(cr, uid, doc_id, model, lang_field, template, values, context=context)
./odoo/addons/report/models/report.py:145:            translate_doc=translate_doc,
./odoo/addons/website_report/report.py:29:    def translate_doc(self, cr, uid, doc_id, model, lang_field, template, values, context=None):
./odoo/addons/website_report/report.py:41:        return super(Report, self).translate_doc(cr, uid, doc_id, model, lang_field, template, values, context=context)
```
And you do not depend of any of them.

We have 3 options here:
1.- You can set in the meta data for the email the  lang with templates and not use this method.
2.- Make this change in an extra module which will be a technical one to extend this capability and depend of website_report.
3. Develop the post mortem test with python and ensure all work as expected with the normal tests on all reports.

On my experience touch this global templating method is too dangerous and bring a lot of unexpected results (I faced some problems already).

Sorry for teh inconvenience here.